monotonic.py

```diff
@@ -0,0 +1,330 @@
+        """
+        This part (lines 80-96) will need improvement - I do not want the data 
+        column #'s hard-coded. Instead, I want them automatically determined by
+        the program.....
+        """
+        #Define Sample Geometry (width and thickness)...
+        #Some systems will not need this step - automatically calc. stress
+        w = mData[10]
+        t = mData[11]
+        w = w[0]
+        t = t[0]
+        Axial_GL = 50 #mm - gauge length of axial extensometer used...
+        Trans_GL = w  #mm
+        Thick_GL = t  #mm
+        #
+        #Define which channels are in which columns...
+        Time = mData[0]      #Time in Seconds
+        Position = mData[6]  #mm (from load frame)
+        Load = mData[7]      #kN (from load cell)
+        Ax_Ext = mData[8]    #% (axial extensometer)
+        Trans_Ext = mData[9] #% (Transverse Extensometer)
+        #        
+        #Necessary Unit Conversions for Analysis...
+        w = w/1000                #mm to m
+        t = t/1000                #mm to m
+        Position = Position/1000  #mm to m
+        Load = Load*1000          #kN to N
+        Axial_GL = Axial_GL/1000  #mm to m
+        Trans_GL = Trans_GL/1000  #mm to m
+        Thick_GL = Thick_GL/1000  #mm to m
+        Ax_Str = Ax_Ext/100       #% to m/m
+        Trans_Str = Trans_Ext/100 #% to m/m
+        #        
+        #Some Calculated Items...
+        A0 = w*t #m^2 - original area
+        extension = Ax_Str*Axial_GL #Axial Extension (m)
+        Trans_mov = Trans_Str*Trans_GL #Transverse change (m)
+        Thick_mov = Trans_Str*Thick_GL #THickness change (m)
+        wi = w - Trans_mov #Instantaneous width
+        ti = t - Thick_mov #Instantaneous thickness
+        CS = wi*ti #Instantaneous cross-section
+        #        
+        #Cut off data at the max of axial strain...
+        """ 
+        Sometimes the extensometer slips at failure and reads a decrease in 
+        strain... we do not want to include this in the analysis so the strain
+        channel (and all other data) is cut off at the index where the maximum 
+        axial strain occurs
+        """
+        max_str = max(Ax_Str)
+        #returns list containing the index where max_str occurs
+        endofdata = np.where(Ax_Str==max_str) 
+        #Pull index out of the list...
+        endofdata = endofdata[-1][0]
+        #Trim all of the data channels...
+        Time = Time[:endofdata]
+        Load = Load[:endofdata]
+        Position = Position[:endofdata]
+        Ax_Str = Ax_Str[:endofdata]
+        Trans_Str = Trans_Str[:endofdata]
+        CS = CS[:endofdata]
+        extension = extension[:endofdata]
+        #        
+        #Define Engineering Stress & Strain...
+        Eng_S = Load/A0 #Pa - Load / Area
+        Eng_E = Ax_Str #m/m - Axial Extensometer reading
+        #        
+        #Calculate True stress & strain ...
+        True_S = Eng_S * (1 + Eng_E) #Pa
+        True_E = np.log(1 + Eng_E)   #m/m

```

Comment:
These could also be separate functions. 
Ex. def define_eng_stress(Load, A0): 
            return Load/A0
    
    def define_eng_strain(Ax_Str): 
           #not: I wouldn't actually make a function that just returns an input. I am just illustrating a point. 
            return Ax_Str 

     def cal_true_stress():
           eng_stress = define_eng_stress(Load, A0)
           eng_strain =  define_eng_strain(Ax_Str)
           return eng_stress * (1+eng_strain)